### PR TITLE
chore(DataList): convert examples to TypeScript 

### LINKED ISSUE #7139: Data list: convert examples to TypeScript
Describe the issue. What is the expected and unexpected behavior?
The examples for the Code editor component need to be converted to TypeScript and functional components.

Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?
Enhancement

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #7139

**Additional issues**:

### CODE REVIEW

File: packages/react-core/src/components/DataList/examples/DataListDraggable.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+interface ItemType {
+  id: string;
+  content: string;
+}
+
+const getItems = count =>
+  Array.from({ length: count }, (v, k) => k).map(k => ({
+    id: `item-${k}`,
+    content: `item ${k} `.repeat(k === 4 ? 20 : 1)
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DataList/examples/DataListDraggable.tsx:24]: Nitpick: may be something worth fixing in another issue, but I think it'd be worth giving these `v` and `k` params more meaningful names

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Only a few comments below, a couple of requested changes and one nitpick. Also the comment I left on the DatePicker conversion PR applies here, so depending on your response to that (regarding the export names from each example) that may apply here, too.

File: packages/react-core/src/components/DataList/examples/DataListExpandable.tsx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+  const [expanded, setExpanded] = React.useState(['ex-toggle1', 'ex-toggle3']);
+
+  const onToggleAll = () => {
+    setAllExpanded(!allExpanded);
+
+    if (allExpanded) {
+      setExpanded(['ex-toggle1', 'ex-toggle2', 'ex-toggle3']);
+    } else {
+      setExpanded([]);
+    }
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply packages/react-core/src/components/DataList/examples/DataListExpandable.tsx:35]: Nice catch! Great idea.

File: packages/react-core/src/components/DataList/examples/DataListWidthModifiers.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+    color: '#004e8a',
+    backgroundColor: '#def3ff',
+    border: '1px solid rgba(0,0,0,.1)',
+    borderRadius: '4px'
+  };
+
+  return (
+    <>
+      <div key="example-1">
+        <h2>Default fitting - example 1</h2>
```
[the remark this answers is not in the document]
Comment (jenny-s51) [reply packages/react-core/src/components/DataList/examples/DataListWidthModifiers.tsx]: 👍 Updated! Definitely looks better - thanks!

**Review by thatblindgeye — APPROVED** [review]
Updates look great!

**Review by nicolethoen — CHANGES_REQUESTED** [review]
Looks great! 

Could you rebase your PR? I think there is a new Data List example that we should be sure we don't leave out of the typescript conversions :)

File: packages/react-core/src/components/DataList/examples/DataList.md [hunk 4]
```diff
@@ -27,1237 +27,49 @@ import { css } from '@patternfly/react-styles';
...
-          <DataListCell key="secondary content">
-            Really really really really really really really really really really really really really really long
-            description that should be truncated before it ends
-          </DataListCell>
-        ]}
-      />
-    </DataListItemRow>
-  </DataListItem>
-</DataList>
+```ts isBeta file="./DataListDraggable.tsx"
```
Comment (tlabaj) [inline packages/react-core/src/components/DataList/examples/DataList.md:74]: Looks like you are missing the `DataListSmGridBreakpoint.tsx` file in the MD file.

**Review by tlabaj — CHANGES_REQUESTED** [review]
You are just missing one file in your MD file.  Looks good otherwise.

File: packages/react-core/src/components/DataList/examples/DataList.md (hunk 4 above)
Comment (jenny-s51) [reply packages/react-core/src/components/DataList/examples/DataList.md:74]: Nice catch @tlabaj - thank you! Updated.

**Review by tlabaj — APPROVED** [review]
LGTM

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**